azurerm/resource_arm_analysis_services_server.go

```diff
@@ -0,0 +1,449 @@
+	}
+
+	return nil
+}
+
+func validateAnalysisServicesServerName(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+
+	if !regexp.MustCompile(`^[a-z][0-9a-z]{2,62}$`).Match([]byte(value)) {
+		errors = append(errors, fmt.Errorf("%q must begin with a letter, be lowercase alphanumeric, and be between 3 and 63 characters in length", k))
+	}
+
+	return warnings, errors
+}
+
+func validateQuerypoolConnectionMode() schema.SchemaValidateFunc {
+	connectionModes := make([]string, len(analysisservices.PossibleConnectionModeValues()))
+	for i, v := range analysisservices.PossibleConnectionModeValues() {
+		connectionModes[i] = string(v)
+	}
+
+	return validation.StringInSlice(connectionModes, true)
+}
+
+func expandAnalysisServicesServerSku(sku string) *analysisservices.ResourceSku {
+	return &analysisservices.ResourceSku{
+		Name: &sku,
+	}
+}
+
+func flattenAnalysisServicesServerSku(sku *analysisservices.ResourceSku) string {
+	return *sku.Name
+}
+
+func expandAnalysisServicesServerProperties(d *schema.ResourceData) *analysisservices.ServerProperties {
+	adminUsers := expandAnalysisServicesServerAdminUsers(d)
+
+	serverProperties := analysisservices.ServerProperties{AsAdministrators: adminUsers}
+
+	//if backupBlobContainerUri, ok := d.GetOk("backup_blob_container_uri"); ok {
+	//	serverProperties.BackupBlobContainerURI = backupBlobContainerUri.(*string)
+	//}
+
+	serverProperties.IPV4FirewallSettings = expandAnalysisServicesServerFirewallSettings(d)
+
+	if connectionMode := expandAnalysisServicesServerQuerypoolConnectionMode(d); connectionMode != nil {
+		serverProperties.QuerypoolConnectionMode = *connectionMode
+	}
+
+	return &serverProperties
+}
+
+func expandAnalysisServicesServerMutableProperties(d *schema.ResourceData) *analysisservices.ServerMutableProperties {
+	adminUsers := expandAnalysisServicesServerAdminUsers(d)
+
+	serverProperties := analysisservices.ServerMutableProperties{AsAdministrators: adminUsers}
+
+	//if backupBlobContainerUri, ok := d.GetOk("backup_blob_container_uri"); ok {
+	//	serverProperties.BackupBlobContainerURI = backupBlobContainerUri.(*string)
+	//}
+
+	serverProperties.IPV4FirewallSettings = expandAnalysisServicesServerFirewallSettings(d)
+
+	if connectionMode := expandAnalysisServicesServerQuerypoolConnectionMode(d); connectionMode != nil {
+		serverProperties.QuerypoolConnectionMode = *connectionMode
+	}
+
+	return &serverProperties
+}
+
+func expandAnalysisServicesServerAdminUsers(d *schema.ResourceData) *analysisservices.ServerAdministrators {
+	adminUsers := d.Get("admin_users").(*schema.Set)
+	members := make([]string, 0)
+
+	for _, admin := range adminUsers.List() {
+		if adm, ok := admin.(string); ok {
+			members = append(members, adm)
+		}
+	}
+
+	return &analysisservices.ServerAdministrators{Members: &members}
+}
+
+func expandAnalysisServicesServerQuerypoolConnectionMode(d *schema.ResourceData) *analysisservices.ConnectionMode {
```

Comment:
I don't think this function is necessary.

Reply:
removed (in case you mean we can do it inline)